src/main/java/org/candlepin/subscriptions/inventory/db/model/InventoryHostFacts.java

```diff
@@ -0,0 +1,148 @@
+import java.util.Set;
+
+/**
+ * Represents an inventory host's facts.
+ */
+public class InventoryHostFacts {
+    private static final Logger log = LoggerFactory.getLogger(InventoryHostFacts.class);
+
+    private String account;
+    private String displayName;
+    private String orgId;
+    private int cores;
+    private int sockets;
+    private boolean isRhel;
+    private String syncTimestamp;
+    private Set<String> products;
+
+    public InventoryHostFacts(String account, String displayName, String orgId, String cores, String sockets,
+        String isRhel, String products, String syncTimestamp) {
+        this.account = account;
+        this.displayName = displayName;
+        this.orgId = orgId;
+        this.cores = asInt(cores);
+        this.sockets = asInt(sockets);
+        this.isRhel = asBoolean(isRhel);
+        this.products = asProducts(products);
+        this.syncTimestamp = StringUtils.hasText(syncTimestamp) ? syncTimestamp : "";
+    }
+
+    public String getAccount() {
+        return account;
+    }
+
+    public void setAccount(String account) {
+        this.account = account;
+    }
+
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    public void setDisplayName(String displayName) {
+        this.displayName = displayName;
+    }
+
+    public String getOrgId() {
+        return orgId;
+    }
+
+    public void setOrgId(String orgId) {
+        this.orgId = orgId;
+    }
+
+    public Integer getCores() {
+        return cores;
+    }
+
+    public void setCores(Integer cores) {
+        this.cores = cores;
+    }
+
+    public Integer getSockets() {
+        return sockets;
+    }
+
+    public void setSockets(Integer sockets) {
+        this.sockets = sockets;
+    }
+
+    public boolean isRhel() {
+        return isRhel;
+    }
+
+    public void setRhel(boolean rhel) {
+        isRhel = rhel;
+    }
+
+    public String getSyncTimestamp() {
+        return syncTimestamp;
+    }
+
+    public void setSyncTimestamp(String syncTimestamp) {
+        this.syncTimestamp = syncTimestamp;
+    }
+
+    public Set<String> getProducts() {
+        return products;
+    }
+
+    public void setProducts(Set<String> products) {
+        this.products = products;
+    }
+
+    private int asInt(String value) {
+        try {
+            return StringUtils.hasText(value) ? Integer.valueOf(value) : 0;
+        }
+        catch (NumberFormatException nfe) {
+            return 0;
+        }
+    }
+
+    private boolean asBoolean(String value) {
+        return StringUtils.hasText(value) && value.equalsIgnoreCase("true");
+    }
+
+    private Set<String> asProducts(String productJson) {
+        if (!StringUtils.hasText(productJson)) {
+            return new HashSet<>();
+        }
+
+        // FIXME Should be able to do this via the query. This code is a GROSS quick
+        //       hack to see if the whole query concept would make a difference. This
+        //       has to go.
```

Comment:
This has to go and is the remaining bit of work left for this PR. There should be a way to produce a comma separated list via the query.